profiles/lib/handlers.lua

```diff
@@ -399,15 +399,22 @@ function Handlers.handle_oneway(way,result,data,profile)
   end
 end
 
+function Handlers.handle_weights(way,result,data,profile)
+  if properties.weight_name == 'distance' then
+    -- set weight rates to 1 for the distance weight, edge weights are distance / rate
+    result.forward_rate = 1
```

Comment:
As @emiltin pointed out this needs a check for either `{foward,backward}_speed > 0` or `{forward,backward}_mode != mode.inaccessible`.